core/common/src/main/java/alluxio/collections/IndexedSet.java

```diff
@@ -305,6 +275,9 @@ public T getFirstByField(FieldIndex<T> index, Object value) {
    */
   @Override
   public boolean remove(Object object) {
+    if (object == null) {
```

Comment:
As I remember, there is a situation that a null object could be passed in after the change. That requires us to do the check. But I don't remember where it is.